aws/delta-checkpoint/src/main.rs

```diff
@@ -0,0 +1,201 @@
+//! Lambda function for writing checkpoints from S3 event triggers on _delta_log JSON files.
+//!
+
+// Reference:
+//
+// https://aws.amazon.com/blogs/opensource/rust-runtime-for-aws-lambda
+// https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-notification-configuration.html
+// https://docs.aws.amazon.com/lambda/latest/dg/with-s3.html
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-filtering.html
+
+use deltalake::checkpoints::{CheckPointWriter, CheckPointWriterError};
+use deltalake::DeltaDataTypeVersion;
+use lambda_runtime::{handler_fn, Context, Error};
+use lazy_static::lazy_static;
+use log::*;
+use regex::Regex;
+use serde_json::Value;
+use std::num::ParseIntError;
+use std::path::PathBuf;
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let func = handler_fn(func);
+    lambda_runtime::run(func).await?;
+    Ok(())
+}
+
+async fn func(event: Value, _: Context) -> Result<(), CheckPointLambdaError> {
+    process_event(&event).await
```

Comment:
Since this is propagating the `CheckPointLambdaError`up to the lambda runtime caller, will that actually put an error message out into the Lambda logs? 

I think it would be good to check, otherwise failures will be completely opaque to the user.

Also, since there is no (pretty) env logger init here, there's no way that i can think of for the log messages to actually get to the Lambda output logs.

Reply:
Ahhh. Errors do show, but my other logs were not. Ty for the pointer to pretty_env_logger. Fixed.